draft-ietf-core-echo-request-tag.md

```diff
@@ -174,7 +174,15 @@ Client   Server
 ~~~~~~~~~~
 {: #echo-figure title="Example Echo Option Message Flow" artwork-align="center"}
 
-When used to serve freshness requirements (including client aliveness and state synchronizing), CoAP messages containing the Echo option MUST be integrity protected between the intended endpoints, e.g. using DTLS, TLS, or an OSCORE Inner option ({{RFC8613}}). When used to demonstrate reachability at a claimed network address, the Echo option SHOULD contain the client's network address, but MAY be unprotected.
+Another way for the server to verify freshness is to maintain a cache of values associated to events: 
+
+* The cache size (defined by the application) is greater than or equal to 1.
+* All cache values MUST be different, or different with a probability close to 1 so they are different for all practical purposes.
+* The oldest value in the cache is overwritten at new events.
+
+The message flow in Figure 2 can serve as illustration also of event-based freshness. In this case t0 is defined as the most recent value in the cache when the server sends the 4.01 (Unauthorized) response. The Echo option value is echoed by the client in the second request and compared by the server with the contents of the cache at the reception of the second request. If t0 is in the cache then the request is considered fresh, otherwise the request is not considered fresh. (In the case of cache size equals 1, t1 is defined as the single value in the cache, and freshness is determined by comparing for equality between t0 and t1.)
```

Comment:
Given we're up to reorder text it'd be useful to have kramdown resolve that number and refer to it by anchor in the source.